### DISCUSSION TITLE
Inserting launchers in unconventional position.

### DISCUSSION THREAD

**Discussion by Diego-planckian1** [iqm-finland/KQCircuits#78] [Q&A] [answered]
Hi guys,

I'm new in KQCircuit python package. I'd like to build my own chip. I'm starting from the launcher insertion only in two side of the chip and also at an height of my choice. Since now, as I understand from the documentation, is possible to change only the number of total launcher and their type. I'd like to use your package in a way that is custimizeble for the chip holder we have. It has not RF connection on each side but only in two of them.

Can you help me?

**Comment by qpavsmi** [comment]
Hi! Unfortunately I'm going to vacation today until 8th of January. We have some configurability with how to adjust your launchers for specific sampleholders, defined in `defaults.py` in `default_sampleholders`. Then `chip.py` can use that configuration to produce the launchers. Since you only need two launchers, you can checkout our qualityfactor chips since that is how we construct them. If you only specify certain launchers in `launcher_assignments` then it will only draw those launchers.

If you need something more custom than that then I would suggest implementing the launcher placement on your own. Feel free to experiment by modifying the all code involving launchers in `chip.py` to get the result you want. It's also good exercise to get to know KQCircuits and KLayout better.

Like I said I can't help more thoroughly until I come back from vacation, but I've notified about your request to my coworker and they might give you some more help before my vacation ends.

**Comment by theandyguthrie** [comment, marked as answer]
Hi Diego,

Thanks for your interest! Yes as Pavel said there are a few ways to proceed. `produce_n_launchers` can take a tuple as an argument for `n` which let's you place just two launchers. I am trying the following code in `chips/quality_factor.py` e.g.: 

`self.produce_n_launchers((0, 1, 0, 1), 'RF', 160, 80, 500, 2000, launcher_assignments={1: "PL-1-OUT", 2: "PL-1-IN"})`

or, if you want them to be at specific locations rather than centered, you could also produce more launchers can you need and only assign the ones you use:

`self.produce_n_launchers((0, 6, 0, 6), 'RF', 160, 80, 500, 1000, launcher_assignments={1: "PL-1-OUT", 7: "PL-1-IN"})`
or
`self.produce_n_launchers((0, 6, 0, 6), 'RF', 160, 80, 500, 1000, launcher_assignments={6: "PL-1-OUT", 12: "PL-1-IN"})`

Or if you would like something more custom you could play with the code in `produce_n_launchers`, but to me it seems this can already solve your issue if you don't mind playing with the numbers in the `launcher_assignments`. 

Let me know if you have any more questions!

**Reply by theandyguthrie** [reply]
![image](https://github.com/iqm-finland/KQCircuits/assets/22938211/bad66acc-8b41-440a-a9ed-553791b83fc4)

![image](https://github.com/iqm-finland/KQCircuits/assets/22938211/4f64cab4-f909-48ac-9b70-7c37c2a288a4)

![image](https://github.com/iqm-finland/KQCircuits/assets/22938211/0d4d559a-92ce-4c74-a231-5038f6d7f701)

**Reply by Diego-planckian1** [reply]
Thank you, I understand how produce_n_launcher works! problem solved
